Libraries/src/Amazon.Lambda.RuntimeSupport/Helpers/ConsoleLoggerWriter.cs

```diff
@@ -112,8 +135,31 @@ enum LogLevel
         /// Stderror will default log messages to be Error
         /// </summary>
         public LogLevelLoggerWriter()
-            : this(Console.Out, Console.Error)
         {
+            // Look to see if Lambda's telemetry log file descriptor is available. If so use that for logging.
+            // This will make sure multiline log messages use a single CloudWatch Logs record.
+            var fileDescriptorLogId = Environment.GetEnvironmentVariable(Constants.ENVIRONMENT_VARIABLE_TELEMETRY_LOG_FD);
+            if (fileDescriptorLogId != null)
+            {
+                try
+                {
+                    var stdOutWriter = FileDescriptorLogFactory.GetWriter(fileDescriptorLogId);
+                    var stdErrorWriter = FileDescriptorLogFactory.GetWriter(fileDescriptorLogId);
+                    Initialize(stdOutWriter, stdErrorWriter);
+                    InternalLogger.GetDefaultLogger().LogInformation("Using file descriptor stream writer for logging.");
```

Comment:
Do this and ConsoleLoggerWriter.cs L53 log line appear same when logged? I believe there should be some prefix to differentiate the log lines emitted from different classes to make debugging easier.

Reply:
There is no precedence for adding a prefix when using InternalLogger.GetDefaultLogger(). If we really wanted to do something like that it would be implemented inside `InternalLogger`.